tiledb/api/c_api/query_aggregate/query_aggregate_api.cc

```diff
@@ -0,0 +1,284 @@
+ */
+inline void ensure_query_channel_is_valid(
+    const tiledb_query_channel_handle_t* channel) {
+  ensure_handle_is_valid(channel);
+}
+
+capi_return_t tiledb_channel_create_operator_sum(
+    tiledb_ctx_t* ctx, tiledb_channel_operator_t** op) {
+  (void)ctx;
+  ensure_output_pointer_is_valid(op);
+  *op = tiledb_channel_operator_handle_t::make_handle(
+      QueryChannelOperator::SUM, "SUM");
+  return TILEDB_OK;
+}
+
+capi_return_t tiledb_channel_create_operator_count(
+    tiledb_ctx_t* ctx, tiledb_channel_operator_t** op) {
+  (void)ctx;
+  ensure_output_pointer_is_valid(op);
+  *op = tiledb_channel_operator_handle_t::make_handle(
+      QueryChannelOperator::COUNT, "COUNT");
+  return TILEDB_OK;
+}
+
+capi_return_t tiledb_query_get_default_channel(
+    tiledb_ctx_t* ctx,
+    tiledb_query_t* query,
+    tiledb_query_channel_t** channel) {
+  (void)ctx;
+  ensure_query_arg_is_valid(query);
+  ensure_output_pointer_is_valid(channel);
+
+  // We don't have an internal representation of a channel,
+  // the default channel is currently just a hashmap, so only pass the query
+  // to the channel constructor to be carried until next the api call.
+  *channel = tiledb_query_channel_handle_t::make_handle(query);
+
+  return TILEDB_OK;
+}
+
+capi_return_t tiledb_channel_create_operation_field(
+    tiledb_ctx_t* ctx,
+    tiledb_query_t* query,
+    const tiledb_channel_operator_t* op,
+    const char* input_field_name,
+    tiledb_channel_operation_t** operation) {
+  (void)ctx;
+  ensure_query_arg_is_valid(query);
+  ensure_channel_operator_is_valid(op);
+  ensure_output_pointer_is_valid(operation);
+
+  shared_ptr<tiledb::sm::IAggregator> aggregator;
+  switch (op->value()) {
+    case QueryChannelOperator::COUNT:
+      aggregator = std::make_shared<tiledb::sm::CountAggregator>();
+      break;
+    case QueryChannelOperator::SUM: {
+      ensure_input_field_is_valid(input_field_name, op->name());
+      std::string field_name(input_field_name);
+
+      auto& schema = query->query_->array_schema();
+      auto g = [&](auto T) {
+        if constexpr (!std::is_same_v<decltype(T), char>) {
+          aggregator = std::make_shared<tiledb::sm::SumAggregator<decltype(T)>>(
+              tiledb::sm::FieldInfo(
+                  field_name,
+                  schema.var_size(field_name),
+                  schema.is_nullable(field_name),
+                  schema.cell_val_num(field_name)));
+        }
+      };
+      execute_callback_with_type(schema.type(field_name), g);
+      break;
+    }
+    default:
+      throw CAPIStatusException(
+          "operator argument `op` has unsupported value: " +
+          std::to_string(static_cast<uint8_t>(op->value())));
+      break;
+  }
+  *operation = tiledb_channel_operation_handle_t::make_handle(aggregator);
+
+  return TILEDB_OK;
+}
+
+capi_return_t tiledb_channel_add_aggregate(
+    tiledb_ctx_t* ctx,
+    tiledb_query_channel_t* channel,
+    const char* output_field_name,
```

Comment:
Output fields are for the query, not just the channel.

Reply:
Would something like `tiledB_add_aggregate_for_channel` make more sense to you? This function is meant to associate an operation with a channel, channel here carries a back pointer to parent query because indeed the mapping between (channel, operations, output field) is owned internally by Query.